tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -435,6 +456,55 @@ def test_harvesting_arxiv_workflow_manual_rejected(
     side_effect=fake_beard_api_block_request,
 )
 @mock.patch(
+    'inspirehep.modules.refextract.tasks.extract_references_from_file',
+    side_effect=fake_refextract_extract_references_from_file,
+)
+def test_harvesting_arxiv_workflow_already_on_legacy(
+    mocked_refextract_extract_refs,
+    mocked_api_request_beard_block,
+    mocked_api_request_magpie,
+    mocked_api_request_beard,
+    mocked_download,
+    small_app,
+    already_harvested_on_legacy_record,
+):
+    """Test a full harvesting workflow."""
+
+    extra_config = {
+        "BEARD_API_URL": "http://example.com/beard",
+        "MAGPIE_API_URL": "http://example.com/magpie",
+    }
+
+    workflow_uuid = None
+    with small_app.app_context():
+        with mock.patch.dict(app.config, extra_config):
+            workflow_uuid = start('article', [
+                already_harvested_on_legacy_record])
+
+        eng = WorkflowEngine.from_uuid(workflow_uuid)
+        obj = eng.processed_objects[0]
+
+        assert obj.status == ObjectStatus.COMPLETED
+        assert obj.extra_data["already-ingested"]
```

Comment:
You might want to check first if the key exists in the `extra_data`, makes nicer errors.